packages/integrations/netlify/src/index.ts

```diff
@@ -116,16 +124,41 @@ async function writeNetlifyFrameworkConfig(config: AstroConfig, logger: AstroInt
 			.filter(Boolean as unknown as (pattern?: string) => pattern is string),
 	);
 
-	const headers = config.build.assetsPrefix
-		? undefined
-		: [
-				{
-					for: `${config.base}${config.base.endsWith('/') ? '' : '/'}${config.build.assets}/*`,
-					values: {
-						'Cache-Control': 'public, max-age=31536000, immutable',
-					},
-				},
-			];
+	const headers = [];
+	if (!config.build.assetsPrefix) {
+		headers.push({
+			for: `${config.base}${config.base.endsWith('/') ? '' : '/'}${config.build.assets}/*`,
+			values: {
+				'Cache-Control': 'public, max-age=31536000, immutable',
+			},
+		});
+	}
+
+	if (staticHeaders && staticHeaders.size > 0) {
+		for (const [route, routeHeaders] of staticHeaders.entries()) {
+			if (!route.isPrerendered) {
+				continue;
+			}
+			if (route.redirect) {
+				continue;
+			}
+
+			const definition = createHostedRouteDefinition(route, config);
+
+			if (config.experimental.csp) {
+				const csp = routeHeaders.get('Content-Security-Policy');
+
+				if (csp) {
+					headers.push({
+						for: definition.input,
+						values: {
+							'Content-Security-Policy': csp,
+						},
+					});
+				}
+			}
```

Comment:
Why do we need CSP-specific handling here? Can we just handle any header that's passed in, and rely on core to decide whether to set the CSP headers?

Reply:
I think we can, yes, once the feature is out of experimental.